tests/e2e/backup_restore_suite_test.go

```diff
@@ -143,9 +142,12 @@ func runBackupAndRestore(brCase BackupRestoreCase, expectedErr error, updateLast
 	nsRequiresResticDCWorkaround, err := NamespaceRequiresResticDCWorkaround(dpaCR.Client, brCase.ApplicationNamespace)
 	Expect(err).ToNot(HaveOccurred())
 
-	// TODO this should be a function, not an arbitrary sleep
-	log.Printf("Sleeping for %v to allow application to be ready for case %s", brCase.AppReadyDelay, brCase.Name)
-	time.Sleep(brCase.AppReadyDelay)
+	if strings.Contains(brCase.Name, "twovol") {
```

Comment:
I like this improvement of removing it from function, but the proper solution would be to something less "around x seconds", how about putting `readinessProbe` in the `tests/e2e/sample-applications/mysql-persistent/mysql-persistent-twovol-csi.yaml` ? 

```
            volumeMounts:
            - name: applog
              mountPath: /tmp/log/todoapp/
            readinessProbe:
              exec:
                command:
                - sh
                - -c
                - '[ -f /tmp/log/todoapp/app.log ]'
              initialDelaySeconds: 1    # Delay before the first probe
              periodSeconds: 5          # Probe interval
              timeoutSeconds: 1         # Maximum time for the probe to complete, should be immediate as it's just file check
              successThreshold: 1       # One time is enough to check for file 
              failureThreshold: 30      # Number of consecutive failed probes to be considered failed   
```

Reply:
That would check, until it succeeds, if the file exists, right?

I think that would not be enough, the problem we face is not that the file does not exist, it is that volume does not have all logs yet

example

first log is saying application started (this one creates files, if not present, I assume)
second log is saying 2 things were added to database (this one, without hardcoded timeout, is not present in backup)

Reply:
So we are checking for the actual item to be in database as well? That could be also part of the check here, just slightly different logic. I am thinking here to solve the issue once and for all to not come back to this problem and actually remove the logic from the go code as that should not be mixed too much. Once the container is ready (whatever we do to make it ready via init-container or probe) then the test should continue. The test itself should not have too much knowledge about application itself.

Reply:
would not that create a cyclic dependency? 
1. we need to check if application is running (also check probes)
2. then we interact with application (creating TO DO items)
3. then we backup

are you not asking to check 2 inside 1, which is impossible?

Reply:
2 is not inside 1. 1. only indicates app is awake and ready to accept commands.